src/controller/CHIPDeviceController.cpp

```diff
@@ -1140,7 +1141,9 @@ CHIP_ERROR DeviceCommissioner::SendOperationalCertificateSigningRequestCommand(D
     Callback::Cancelable * successCallback = mOpCSRResponseCallback.Cancel();
     Callback::Cancelable * failureCallback = mOnCSRFailureCallback.Cancel();
 
-    ReturnErrorOnFailure(device->GenerateCSRNonce());
+    if(!device->IsCSRNonceProvided()){
```

Comment:
This block can be deleted if you generate or set the `CSRNonce` in the `PairDevice()`

Reply:
Done